src/analyzer.rs

```diff
@@ -69,17 +69,17 @@ impl<'run, 'src> Analyzer<'run, 'src> {
           } => {
             let mut doc_attr: Option<&str> = None;
             let mut groups = Vec::new();
-            for attribute in attributes {
+            attributes.ensure_valid_attributes(
```

Comment:
I think we should keep the error checking in the `else` below, otherwise the set of allowed attributes is in two places, here, where we call `ensure_valid_attributes`, and below, in the `if / else` statement.

Reply:
If we keep the error checking here, then we would have pretty similar error-generation code here and in the `ensure_valid_attributes` method. I added `ensure_valid_attributes` as a way of de-duplicating attribute error generation code. I modified the code below to add an `unreachable` and hopefully make it clearer what's going on.